Documentation/install/upgrade.rst

```diff
@@ -320,6 +320,18 @@ IMPORTANT: Changes required before upgrading to 1.8.0
   .. note:: The ENI IPAM mode automatically derives the native routing CIDR so
             no action is required.
 
+Upgrading from >=1.7.0 to 1.8.y
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Since Cilium 1.5, the conntrack table size ``ct-global-max-entries-tcp``
```

Comment:
This is tricky to get right. In general, we are providing instructions to users with the Helm options since v1.6, so the option here may be confusing because if someone applies it in Helm. For example, `--set ct-global...` it will have no effect.

I would suggest that instead we have a two-phase process:

1. Identify whether any action is necessary on the user's part. This could be either getting the logs from Cilium startup and checking the value (unreliable for long-running instances) or maybe using something like `bpftool` inside a cilium container to identify what the current size of the map is. We can likely streamline this to a 1-liner.
1. If (and only if) the value is non-standard, provide the helm options necessary to ensure that the `helm template` / `helm install` commands will generate the correct configuration for these settings.

Reply:
One other caveat here -- I didn't trace through the various install instructions but what subset of users will be affected here? I think we can safely say that:
* Users who installed v1.5, v1.6 or v1.7 on a fresh cluster and subsequently upgrade will not be affected.
* Users who upgrade to newer Cilium versions while retaining their old ConfigMap (option B [here](https://docs.cilium.io/en/v1.7/install/upgrade/#step-2-option-b-preserve-configmap) *may* be affected (but it would depend on the initial version of Cilium used and the YAMLs provided with that version, and whether the user has made any manual changes to the ConfigMap since then..).

I would encourage you to investigate these statements and disagree if you find evidence that they're untrue.

The importance of this is that we can frame the documentation in such a way that we can potentially simplify the upgrade process for a plurality of users. Two focuses: (1) as a user, how much do you care about hitless upgrade / connection interruption (if you don't care, ignore these instructions) ; (2) how can a user know easily that they must take action (if you care, do you need to do something).

Reply:
Thanks a lot for the feedback @joestringer 

re. your 1st reply: I added the corresponding instructions to determine the current map size (using the `bpf-map` tool, as `bpftool` didn't seem to work for me on 4.9 kernel). Also added the instructions to set the helm options.

re. your 2nd reply: I traced through the install and upgrade instructions and as far as I can see, your statements given above are all true. But to be honest I'm not yet that familiar with helm and ConfigMap, so I might have missed some subtle detail.

> The importance of this is that we can frame the documentation in such a way that we can potentially simplify the upgrade process for a plurality of users. Two focuses: (1) as a user, how much do you care about hitless upgrade / connection interruption (if you don't care, ignore these instructions) ; (2) how can a user know easily that they must take action (if you care, do you need to do something).

Completely agree. I think having the instructions with the check for the currently configured map size should help in that regard. Maybe it can be fleshed out a bit more to be more concise. Feedback welcome.

Reply:
@joestringer waiting for an ack on your side before I merge this <message from janitor hat>